src/MigrateStructure.php

```diff
@@ -262,10 +262,30 @@ function (GrantToRole $v) use ($database, $schemaName, $tableName) {
         }
     }
 
-    public function createMainRole(Role $mainRoleWithGrants): void
+    public function createMainRoleAndUser(Role $mainRoleWithGrants): void
     {
         $user = $mainRole = $mainRoleWithGrants->getName();
 
+        $mainRoleExists = $this->destinationConnection->fetchAll(sprintf(
+            'SHOW ROLES LIKE %s',
+            QueryBuilder::quote($mainRole),
+        ));
```

Comment:
Pro tyhle use cases by se hodil `Keboola\SnowflakeDbAdapter\Connection::count)` 🙂 

```suggestion
        $mainRoleExists = count($this->destinationConnection->fetchAll(sprintf(
            'SHOW ROLES LIKE %s',
            QueryBuilder::quote($mainRole),
        ))) > 0;
```

Reply:
jo asi může být